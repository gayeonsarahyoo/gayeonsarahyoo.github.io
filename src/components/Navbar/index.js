import React from 'react';
import * as ROUTES from '../../constants/routes';

class Navbar extends React.Component {
  constructor(props){
		super(props)
		this.state = { loggedinstate: '', logoutbutton: null, loginbutton: null, signupbutton: null}
		this.isLoggedIn = this.isLoggedIn.bind(this)
	}
	/*
	isLoggedIn() {
		fetch('/user-status').then(res => res.json()).then(data => {
			this.setState({ users: data });
		});
	}*/

	componentDidMount(){
		console.log("nav bar loading");
		this.isLoggedIn();
		
	}

	async saveLogout() {
		fetch('/logout/',{
			method: 'POST',
			headers: {
				'Content-Type': 'application/json'
			}
		});
}

	isLoggedIn = () => {
		fetch('/login-status/')
		.then((response) => response.json())
		.then((data) => this.setState({loggedinstate: data}));
		
}

renderDropdown(){
	if(this.state.loggedinstate['logged in']){
		
		return(
			<div className='navbar-nav ml-auto'>

<ul className="navbar-nav ml-auto">
				 <li className="nav-item dropdown ml-auto">
				<a className="nav-link dropdown-toggle" href="#" id="navbardrop" data-toggle="dropdown">
					Hello {this.state.loggedinstate['name']}!</a>
			<div className="dropdown-menu">
				<a className="dropdown-item" href={ROUTES.PROFILE}>Profile</a>
				<a className="dropdown-item" onClick={() => this.saveLogout()} href={ROUTES.LANDING}>Sign Out</a>
			</div>
		</li>
				 </ul>
			</div>
		);

	}
	else{
		return(
			<div className='navbar-nav ml-auto'>
				<ul className="navbar-nav ml-auto">
				 <li className="nav-item dropdown">
				<a className="nav-link dropdown-toggle" href="#" id="navbardrop" data-toggle="dropdown">
					Welcome</a>
			<div className="dropdown-menu">
				<a className="dropdown-item" href={ROUTES.SIGN_UP}>Sign Up</a>
				<a className="dropdown-item" href={ROUTES.SIGN_IN}>Sign In</a>
			</div>
		</li>
				 </ul>
			</div>
		);

	}
}


  render() {
    return (
     <nav className="navbar navbar-expand-lg navbar-dark bg-dark">
       <a className="navbar-brand" href="#">Aggregate COVID-19 Forecasting</a>
       <button className="navbar-toggler" type="button" data-toggle="collapse" data-target="#navbarNavAltMarkup" aria-controls="navbarNavAltMarkup" aria-expanded="false" aria-label="Toggle navigation">
         <span className="navbar-toggler-icon"></span>
       </button>
       <div className="collapse navbar-collapse" id="navbarNavAltMarkup">
       <div className="navbar-nav">
         <a className="nav-item nav-link active" href={ROUTES.LANDING}>Home <span className="sr-only">(current)</span></a>
         <a className="nav-item nav-link" href={ROUTES.ABOUT}>About</a>
         <a className="nav-item nav-link" href={ROUTES.LEADERBOARD}>Top Forecasts</a>
        
       </div>
       {this.renderDropdown()}
       </div>
     </nav>
    );
  }
}

export default Navbar;
