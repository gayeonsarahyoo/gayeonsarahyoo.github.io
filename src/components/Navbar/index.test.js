import React from 'react';
import Navbar from './';
import renderer from 'react-test-renderer';

test('Navbar renders correctly', () => {
  const component = renderer.create(<Navbar />);
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
