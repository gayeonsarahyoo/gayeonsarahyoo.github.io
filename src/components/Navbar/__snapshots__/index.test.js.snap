// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar renders correctly 1`] = `
<nav
  class="navbar navbar-expand-lg navbar-dark bg-primary"
>
  <a
    class="navbar-brand"
    href="#"
  >
    Navbar
  </a>
  <button
    aria-controls="navbarNavAltMarkup"
    aria-expanded="false"
    aria-label="Toggle navigation"
    class="navbar-toggler"
    data-target="#navbarNavAltMarkup"
    data-toggle="collapse"
    type="button"
  >
    <span
      class="navbar-toggler-icon"
    />
  </button>
  <div
    class="collapse navbar-collapse"
    id="navbarNavAltMarkup"
  >
    <div
      class="navbar-nav"
    >
      <a
        class="nav-item nav-link active"
        href="#"
      >
        Home 
        <span
          class="sr-only"
        >
          (current)
        </span>
      </a>
      <a
        class="nav-item nav-link"
        href="#"
      >
        About
      </a>
    </div>
  </div>
</nav>
`;
